packages/financial-templates-lib/src/price-feed/InsuredBridgePriceFeed.ts

```diff
@@ -0,0 +1,172 @@
+import Web3 from "web3";
+import { parseAncillaryData } from "@uma/common";
+import { BN } from "../types";
+import type { Logger } from "winston";
+import { InsuredBridgeL1Client, Relay } from "../clients/InsuredBridgeL1Client";
+import { InsuredBridgeL2Client, Deposit } from "../clients/InsuredBridgeL2Client";
+
+const { toBN, toWei } = Web3.utils;
+
+enum isRelayValid {
+  No, // Should be 0
+  Yes, // Should be 1
+}
+
+interface Params {
+  logger: Logger;
+  web3: Web3;
+  l1Client: InsuredBridgeL1Client;
+  l2Client: InsuredBridgeL2Client;
+}
+
+// Allows user to respond to a "relay" price request that was sent in response to a "deposit" on a InsuredBridge
+// deployed to an L2 network. The relay price request is submitted on L1. This pricefeed will respond "Yes" or "No"
+// based on whether the relay was correctly constructed to match a deposit. The price request includes parameters in
+// its ancillary data that must be parsed from the hex ancillary data.
+export class InsuredBridgePriceFeed extends PriceFeedInterface {
+  private readonly decimals: number;
+  private readonly logger: Logger;
+  private readonly web3: Web3;
+  private readonly l1Client: InsuredBridgeL1Client;
+  private readonly l2Client: InsuredBridgeL2Client;
+  private relays: Relay[] = [];
+  private deposits: Deposit[] = [];
+  private readonly toBNWei: (_number: number) => BN;
+
+  /**
+   * @notice Constructs the InsuredBridgePriceFeed.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider to connect to Ethereum network.
+   * @param {Object} l1Client Fetches and returns latest state of L1 pool and admin contracts.
+   * @param {Object} l2Client Fetches and returns latest state of L2 deposit contract.
+   */
+  constructor({ logger, web3, l1Client, l2Client }: Params) {
+    super();
+
+    this.decimals = 18;
+    this.logger = logger;
+    this.web3 = web3;
+    this.l1Client = l1Client;
+    this.l2Client = l2Client;
+    this.toBNWei = (_number) => toBN(toWei(_number.toString()));
+  }
+
+  // This method returns the validity of a relay price request attempt. The relay request was valid if and only if it:
+  // (1) corresponds with an L1 relay transaction submitted to the appropriate BridgePool contract and
+  // (2) the L1 relay corresponds with an L2 deposit transaction submitted to the appropriate DepositBox contract.
+  // If the relay request does not meet these conditions, then this method will return a price of 0, implying "No, the
+  // relay was not valid" and the relay action will be disputed.
+
+  // For example, if a malicious actor were to submit a price request directly to the Optimistic
+  // Oracle using the Insured Bridge identifier, then this method would return a price of 0 since there was no
+  // associated relay for the price request.
+  public async getHistoricalPrice(time: number | string, verbose: boolean, ancillaryData: string): Promise<BN> {
```

Comment:
nit: should we move verbose to the end since it can often be ignored? This may induce a bit more churn, but might be a bit cleaner.

Other nit: isn't verbose also optional (shouldn't it be `verbose?: boolean` ?).

Reply:
Yeah this is future technical debt if we don't move verbose all the way to the end I'll change that. Regarding the other nit, i'll remove `verbose` from this function since its no longer neccessary to define.